extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -366,6 +366,31 @@ export class AmpStoryPage extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Installs the effect on the page if the there is no opt-out detected
+   * @private
+   * @return {!Promise}
+   */
+  maybeInitializeParallaxFx_() {
+    return Promise.resolve().then(() => {
+      if (this.element.hasAttribute('no-parallax-fx')) {
+        return;
+      }
+
+      const storyEl = dev().assertElement(
+        closestAncestorElementBySelector(this.element, 'amp-story'),
+        'amp-story-page must be a descendant of amp-story.'
+      );
+
+      return storyEl.getImpl().then(storyImpl => {
```

Comment:
Instead of nesting a `.then` here, it is cleaner to do:

```javascript
  return storyEl.getImpl();
}).then(storyImpl => {
  if (!storyImpl.parallaxManager) {
    return;
  }
  return storyImpl.parallaxManager.registerParallaxPage(this.element);
});